metaflow/plugins/aip/argo_client.py

```diff
@@ -83,6 +83,20 @@ def create_workflow_config_map(self, name: str, config_map: Dict[str, Any]):
                     json.loads(e.body)["message"] if e.body is not None else e.reason
                 )
 
+    def list_workflow_template(self, namespace: Optional[str] = None):
```

Comment:
Could this be renamed to `get_workflow_templates` or `get_workflow_templates_list`? Having the `get` prefix makes it clear what it does.

Reply:
The corresponding argo cli commands is `argo template list` so `list` should be the right prefix. How about I rename all the functions according to argo cli commands? I'm thinking that I create a class `ArgoHelper` with static methods. So the new names can be:

- `ArgoHelper.template_submit` (was `run_argo_workflow`, borrowing the `argo submit` command name, but instead of a workflow we are "submitting a template" that's automatically wrapped in a workflow)
- `ArgoHelper.template_list` (was `list_workflow_template`)
- `ArgoHelper.template_get` (was `get_latest_workflow` , aligned with `argo template get` command)
- `ArgoHelper.template_delete` (was `delete_argo_workflow`, but it's actually deleting a workflow template in the code)
- `ArgoHelper.watch` (was `wait_for_argo_run_completion`)

Reply:
Functions in `argo_utils.py` renamed. The ones in `argo_client.py` seems to be in good shapes already. `list_workflow_template` aligns with argo cli command names.

Reply:
@talebzeghmi Thanks for the concern. There might not be as good a time to refactor after v4 migration is completed due to customer adoptions, so I'm leaning towards a refactor now.